### PR TITLE
feat: playground accepts multiple modalities for query

### PR DESCRIPTION
**Pull request by IyadhKhalfallah**
---

- [ ] This PR references an open issue
- [ ] Test added
- [ ] Documentation adjusted

### CODE REVIEW

**Comment by IyadhKhalfallah** [comment]
v0
Screenshot 2023-01-18 at 15 28 09

File: deployment/playground/playground.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def deploy_streamlit():
 
         filter_selection = {}
         if st.session_state.filters != 'notags':
-            st.sidebar.title('Filters')
```
Comment (azayz) [inline deployment/playground/playground.py:124]: why the , I guess it's a typo

Comment (IyadhKhalfallah) [reply deployment/playground/playground.py]: Nice catch, removed 👍

File: deployment/playground/playground.py [hunk 2]
```diff
@@ -287,60 +295,42 @@ class UI:
...
-        for doc, col in zip(da_txt, [c1, c2, c3, c4, c5, c6]):
-            with col:
-                if st.button(doc.content, key=doc.id, on_click=clear_text):
-                    st.session_state.matches = search_by_text(
-                        search_text=doc.content,
-                        jwt=st.session_state.jwt_val,
-                        filter_selection=filter_selection,
-                    )
+    else:
+        for x in range(st.session_state[f"len_{modality}_choices"]):
```
Comment (azayz) [inline deployment/playground/playground.py:316]: can you find a different naming for x

File: deployment/playground/playground.py [hunk 3]
```diff
@@ -287,60 +295,42 @@ class UI:
...
+            key = f'{modality}_{x}'
+            q = st.file_uploader(
```
Comment (azayz) [inline deployment/playground/playground.py:318]: same for q here

File: deployment/playground/playground.py [hunk 4]
```diff
@@ -287,60 +295,42 @@ class UI:
...
-                        document=doc,
-                        jwt=st.session_state.jwt_val,
-                        filter_selection=filter_selection,
-                    )
-
+def render_mm_query(query, modality):
+    if st.button("+", key=f'{modality}'):
+        st.session_state[f"len_{modality}_choices"] += 1
+    if modality == 'text':
+        for x in range(st.session_state[f"len_{modality}_choices"]):
```
Comment (azayz) [inline deployment/playground/playground.py:302]: what's x here

**Review by azayz — APPROVED** [review]
LGTM after comments

**Review by jupyterjazz — APPROVED** [review]
Nice! 🚀 
can you put the screenshot of the new version too?

**Comment by makram93** [comment]
Also, the search is cleared as soon as I change the value in text/image field. Can we keep the results and only clear it when `search` button is clicked?

File: deployment/playground/playground.py [hunk 5]
```diff
@@ -287,60 +295,42 @@ class UI:
...
-            with txt:
-                if st.button('Search', key=doc.id, on_click=clear_match):
-                    st.session_state.matches = search_by_image(
-                        document=doc,
-                        jwt=st.session_state.jwt_val,
-                        filter_selection=filter_selection,
-                    )
-
+def render_mm_query(query, modality):
+    if st.button("+", key=f'{modality}'):
```
[the remark this answers is not in the document]
Comment (IyadhKhalfallah) [reply deployment/playground/playground.py:299]: @makram93 there is no clear solution on how to perform that (removing a specific input) in streamlit. I can add it to another ticket and research it. What do you think?

**Comment by IyadhKhalfallah** [comment]
> d as soon as I cha

Could you please elaborate more with an example? I couldn't get this point clear

**Comment by IyadhKhalfallah** [comment]
> Also, the search is cleared as soon as I change the value in text/image field. Can we keep the results and only clear it when `search` button is clicked?

Could you please elaborate more with an example? I couldn't get this point clear

File: deployment/playground/playground.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (makram93) [reply deployment/playground/playground.py:299]: I think we can always assume that on deleting we delete the bottom row fields. It is also intuitive

**Pull request merged**